test/gateways/authorize_net_test.exs

```diff
@@ -64,4 +80,120 @@ defmodule Gringotts.Gateways.AuthorizeNetTest do
     end
   end
 
+  describe "authorize" do
+    test "successful response" do
+      with_mock HTTPoison,
+      [request: fn(_method, _url, _body, _headers) -> MockResponse.successful_authorize_response end] do
+        {:ok, response} = ANet.authorize(@amount, @card, @opts)
+        assert response.raw["createTransactionResponse"]["messages"]["resultCode"] == "Ok"
+    end
+    end
+
+    test "with bad amount" do
+      with_mock HTTPoison,
+      [request: fn(_method, _url, _body, _headers) -> MockResponse.bad_amount_purchase_response end] do
+        {:error, response} = ANet.authorize(@bad_amount, @card, @opts)
+        assert response.raw["createTransactionResponse"]["messages"]["resultCode"] == "Error"
+      end
+    end
+
+    test "with bad card" do
+      with_mock HTTPoison,
+        [request: fn(_method, _url, _body, _headers) -> MockResponse.bad_card_purchase_response end] do
+          {:error, response} = ANet.authorize(@amount, @bad_card, @opts)
+          assert response.raw["ErrorResponse"]["messages"]["resultCode"] == "Error"
+      end
+    end
+  end
+
+  describe "capture" do
+    test "successful response" do
+      with_mock HTTPoison,
+      [request: fn(_method, _url, _body, _headers) -> MockResponse.successful_capture_response end] do
+        {:ok, response} = ANet.capture(@amount, @bad_card, @opts)
+        assert response.raw["createTransactionResponse"]["messages"]["resultCode"] == "Ok"
+      end
+    end
+    
+    test "with bad transaction id" do
+      with_mock HTTPoison,
+      [request: fn(_method, _url, _body, _headers) -> MockResponse.bad_id_capture end] do
+        {:error, response} = ANet.capture(@bad_amount, @card, @opts)
+        assert response.raw["createTransactionResponse"]["messages"]["resultCode"] == "Error"
+      end
+    end
+  end
+
+  describe "refund" do
+    test "successul response" do
+      with_mock HTTPoison,
+      [request: fn(_method, _url, _body, _headers) -> MockResponse.successful_refund_response end] do
+        {:ok, response} = ANet.refund(@amount, @refund_id, @opts_refund)
+        assert response.raw["createTransactionResponse"]["messages"]["resultCode"] == "Ok"
+      end
+    end
+
+    test "bad payment params" do
+      with_mock HTTPoison,
+        [request: fn(_method, _url, _body, _headers) -> MockResponse.bad_card_refund end] do
+          {:error, response} = ANet.refund(@amount, @refund_id, @opts_refund)
```

Comment:
The call `ANet.refund(@amount, @refund_id, @opts_refund)`
is no different from the one in successful response (line 131). Why would the response from ANet be different?